src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Layers/ChangeStretchRenderer/ChangeStretchRenderer.xaml

```diff
@@ -1,39 +1,54 @@
 ﻿<UserControl x:Class="ArcGISRuntime.WPF.Samples.ChangeStretchRenderer.ChangeStretchRenderer"
              xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
              xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
-             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" 
-             xmlns:d="http://schemas.microsoft.com/expression/blend/2008" 
-             xmlns:esri="http://schemas.esri.com/arcgis/runtime/2013" 
-             mc:Ignorable="d" 
+             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+             xmlns:esri="http://schemas.esri.com/arcgis/runtime/2013"
+             mc:Ignorable="d"
              d:DesignHeight="300" d:DesignWidth="300">
     <Grid>
         <esri:MapView x:Name="MyMapView" />
         <Border
-                Background="White" BorderBrush="Black" BorderThickness="1"
-                HorizontalAlignment="Right" VerticalAlignment="Top"
-                Margin="30" Width="375">
-            <Grid Margin="20">
+            Background="White" BorderBrush="Black" BorderThickness="1"
+            HorizontalAlignment="Right" VerticalAlignment="Top"
+            Margin="30" Padding="20" Width="375">
+            <Grid>
                 <Grid.RowDefinitions>
                     <RowDefinition />
                     <RowDefinition />
                     <RowDefinition />
                     <RowDefinition />
                 </Grid.RowDefinitions>
-                <TextBlock Text="Choose a stretch renderer type from the dropdown, adjust the parameter values, then click the 'Update Renderer' button." 
-                           Margin="0,0,0,2" TextWrapping="Wrap" HorizontalAlignment="Center" VerticalAlignment="Center" />
+                <TextBlock
+                    Text="Choose a stretch renderer type from the dropdown, adjust the parameter values, then click the 'Update Renderer' button."
+                    Margin="0,0,0,2"
+                    TextWrapping="Wrap"
+                    HorizontalAlignment="Center" VerticalAlignment="Center" />
                 <StackPanel Grid.Row="1">
-                    <ComboBox x:Name="RendererTypes" Margin="10,5" SelectionChanged="RendererTypes_SelectionChanged"/>
+                    <ComboBox x:Name="RendererTypes"
+                              Margin="10,5"
+                              SelectionChanged="RendererTypes_SelectionChanged" />
                     <StackPanel Orientation="Horizontal">
-                        <TextBlock x:Name="Label_Parameter1" Text="Parameter1" Margin="10,5"></TextBlock>
-                        <TextBox x:Name="Input_Parameter1" Text="EnterValue1" />
+                        <TextBlock x:Name="FirstParameterLabel"
```

Comment:
Why not use `NumericUpDown`?

![2018-06-06_11-02-40_arcgisruntime](https://user-images.githubusercontent.com/587809/41056402-e992ec16-6978-11e8-9b46-035e8c39e9dd.png)

Reply:
I had the same thought. NumericUpDown is a winforms control and there is no baked in WPF alternative. There are toolkits and the like, but we've been shy about adding dependencies in the past.

Reply:
Oops, you're right, I always forget... I guess we'd have to roll our own event-based validation in WPF, won't we. Probably not worth the effort.